docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+will be stopped at connection gater (see below).
+
+#### Connection Gating
+
+Connection Gating allows safeguarding against bad/pruned peers that tries to reconnect multiple times. 
+Inbound and outbound connections are intercepted and being checked before other components process the connection.
+
+See libp2p's [ConnectionGater](https://github.com/libp2p/go-libp2p-core/blob/master/connmgr/gater.go) 
+interface for more info.
+
+### Forks
+
+Future network forks will follow the general forks mechanism and design in SSV. \
+The idea is to wrap procedures that have potential to be changed in future versions.
+
+Currently, the following are covered:
+
+- validator topic mapping
+- message encoding/decoding
+- user agent
+
+#### Fork v0
+
+**validator topic mapping**
+
+Validator public key is used as the topic name:
+
+`bloxstaking.ssv.<hex(validator-public-key)>`
+
+**message encoding/decoding**
+
+JSON is used for encoding/decoding of messages.
+
+**user agent**
+
+User Agent contains the node version and type, and in addition the operator id.
+
+`SSV-Node:v0.x.x:<node-type>:<?operator-id>`
+
+#### Fork v1 (TBD)
+
+**validator topic mapping**
+
+Validator public key hash is used to determine the validator's subnet which is the topic name:
+
+`bloxstaking.ssv.<hash(validatiorPubKey) % num_of_subnets>`
+
+### NAT port map
+
+libp2p enables to configure a `NATManager` that will attempt to open a port in the network's firewall using `UPnP`.
+
+
+### Security
+
+The discovery system is naturally a good candidate for security problems. \
+DiscV5 specs specifies potential vulnerabilities in their system, 
+See (DiscV5 Rationale > security-goals)[https://github.com/ethereum/devp2p/blob/master/discv5/discv5-rationale.md#security-goals].
+The major ones includes routing table pollution, traffic redirection, spamming or replayed messages. \
+Some are less relevant to `SSV.Network` as messages are being verified in a higher level, 
+by the IBFT components using the share public key, 
+and therefore malicious messages will be identified as invalid.
+
+The following measures are used to protect against malicious peers and denial of service attacks:
+- Connection score is determined during discovery process, ensures that the node will try to connect only with relevant peers.
+- Connection filters determine the score for inbound connections, and terminates low scored connections
+- Connection gater protects against peers which were pruned in the past and tries to reconnect again before backoff timeout (5 min). 
+it kicks in in an early stage, before the other components processes the request to avoid resources consumption.
+
+
+### High Availability
+
+As it participates in a decentralized p2p network, HA for an ssv node is not trivial. \ 
+The reason is that running multiple instances might lead to slashing or 
+disturb the consensus as it creates ambiguity and conflicts.
+
+Ideas TBD:
+
+#### Hub Node
```

Comment:
I think simply enabling an operator admin key to register multiple operators is an easier solution no?